src/yaml-support/yaml-schema.ts

```diff
@@ -0,0 +1,165 @@
+import * as _ from 'lodash';
+import Uri from 'vscode-uri';
+import * as vscode from 'vscode';
+import { yamlLocator } from "./yaml-locator";
+import { VSCODE_YAML, KUBERNETES_SCHEMA, KUBERNETES_GROUP_VERSION_KIND, GROUP_VERSION_KIND_SEPARATOR } from "./yaml-constant";
+
+// the schema for kubernetes
+const _definitions = {};
+
+// load the kubernetes schema and make some modifications to $ref node
+export function loadSchema(schemaFile: string) {
+    const schemaRaw = require(schemaFile);
+    const definitions = schemaRaw.definitions;
+    const toFixed = [];
+    for (let key of Object.keys(definitions)) {
+        let node = definitions[key];
+        const name = key;
+        if (node[KUBERNETES_GROUP_VERSION_KIND]) {
+            let groupKindNode = node[KUBERNETES_GROUP_VERSION_KIND];
+            delete node[KUBERNETES_GROUP_VERSION_KIND];
+            if (groupKindNode.length) {
+                groupKindNode.forEach((groupKindNodeItem) => {
+                    const apiVersion = ((groupKindNodeItem.Group ? groupKindNodeItem.Group + '/' : '') + groupKindNodeItem.Version);
+                    const kind = groupKindNodeItem.Kind;
+                    _definitions[apiVersion + GROUP_VERSION_KIND_SEPARATOR + kind] = _definitions[name.toLowerCase()] = {
+                        apiVersion,
+                        name,
+                        kind,
+                        ...node
+                    };
+
+
+                });
+            }
+        } else {
+            _definitions[name.toLowerCase()] = {
+                name,
+                ...node
+            };
+        }
+
+        toFixed.push(node['properties']);
+    }
+
+    for (let fix of toFixed ) {
+        patchOnRef(fix);
+    }
+}
+
+export async function registerYamlSchemaSupport() {
+    loadSchema(KUBERNETES_SCHEMA);
+    let yamlPlugin;
+    const installedExtensions: any[] = vscode.extensions.all;
+    for (let ext of installedExtensions) {
+        if (ext.id === VSCODE_YAML) {
```

Comment:
Do we need to do the loop/if check?  Can we not use `vscode.extensions.getExtension`?